src/Workspaces/VisualBasic/Portable/LanguageServices/VisualBasicSyntaxFactsService.vb

```diff
@@ -1312,6 +1312,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                       TryCast(node, MemberAccessExpressionSyntax)?.Name)
         End Function
 
+        Public Function GetLeftSideOfDot(node As SyntaxNode, Optional allowImplicitTarget As Boolean = False) As SyntaxNode Implements ISyntaxFactsService.GetLeftSideOfDot
+            Return If(TryCast(node, QualifiedNameSyntax)?.Left,
+                      TryCast(node, MemberAccessExpressionSyntax)?.GetExpressionOfMemberAccessExpression(allowImplicitTarget))
```

Comment:
so, a cleaner way to do this would be to make GetLeftSideOfDot an extension on ISyntaxFactsService.  It can interrogate IsQualifiedName and IsMemberAccess, and use the GetPartsOfXXX helpers.

Then you don't need a C# and VB version of the identical cod.e